src/pages/trade/ui/route-book.tsx

```diff
@@ -1,60 +1,166 @@
-          </tr>
-        </thead>
-        <tbody>
-          {multiHops.sell.map(trace => (
-            <tr key={trace.price + trace.total} style={{ color: 'red' }}>
-              <td className='text-left tabular-nums'>{trace.price}</td>
-              <td className='text-right tabular-nums'>{trace.amount}</td>
-              <td className='text-right tabular-nums'>{trace.total}</td>
-              <td className='text-right tabular-nums'>{trace.hops.length}</td>
-            </tr>
-          ))}
-          {multiHops.buy.map(trace => (
-            <tr key={trace.price + trace.total} style={{ color: 'green' }}>
-              <td className='text-left tabular-nums'>{trace.price}</td>
-              <td className='text-right tabular-nums'>{trace.amount}</td>
-              <td className='text-right tabular-nums'>{trace.total}</td>
-              <td className='text-right tabular-nums'>{trace.hops.length}</td>
-            </tr>
-          ))}
-        </tbody>
-      </table>
+    <div className='flex flex-col max-w-full border-y border-border-base'>
+      <div className='flex items-center gap-2 px-4 h-11 border-b border-border-base'>
+        <Tabs
+          value={activeTab}
+          onChange={setActiveTab}
+          options={ROUTEBOOK_TABS}
+          actionType='accent'
+        />
+      </div>
+      <div className='flex-1'>
+        {activeTab === 'routes' ? (
+          <table className='w-full'>
+            <thead>
+              <tr className='text-xs text-gray-400'>
+                <th className='py-2 font-normal text-left'>Price({pair.quoteSymbol})</th>
+                <th className='py-2 font-normal text-right'>Amount({pair.baseSymbol})</th>
+                <th className='py-2 font-normal text-right'>Total</th>
+                <th className='py-2 font-normal text-right'>Route</th>
+              </tr>
+            </thead>
+
+            <tbody className='relative'>
+              {multiHops ? (
+                <>
+                  {multiHops.sell.map((trace, idx) => (
+                    <TradeRow
+                      key={`${trace.price}-${trace.total}-${idx}`}
+                      trace={trace}
+                      isSell={true}
+                      relativeSize={sellRelativeSizes.get(trace.total) ?? 0}
+                    />
+                  ))}
+
+                  <SpreadRow sellOrders={multiHops.sell} buyOrders={multiHops.buy} />
+
+                  {multiHops.buy.map((trace, idx) => (
+                    <TradeRow
+                      key={`${trace.price}-${trace.total}-${idx}`}
+                      trace={trace}
+                      isSell={false}
+                      relativeSize={buyRelativeSizes.get(trace.total) ?? 0}
+                    />
+                  ))}
+                </>
+              ) : (
+                Array(17)
+                  .fill(1)
+                  .map((_, i) => <SkeletonRow isSpread={i === 8} key={i} />)
+              )}
```

Comment:
question: we aren't using the isLoading var anymore, are we covering all user paths? 🤔 or maybe we can rely on an `undefined` for data?

Reply:
undefined for data is fine, since we are using it in place of isLoading. for catching errors, we have line 120.